src/sentry/integrations/discord/message_builder/issues.py

```diff
@@ -0,0 +1,149 @@
+    def __init__(
+        self,
+        group: Group,
+        event: GroupEvent | None = None,
+        tags: set[str] | None = None,
+        rules: list[Rule] | None = None,
+        link_to_event: bool = False,
+        issue_details: bool = False,
+        notification: ProjectNotification | None = None,
+    ) -> None:
+        self.group = group
+        self.event = event
+        self.tags = tags
+        self.rules = rules
+        self.link_to_event = link_to_event
+        self.issue_details = issue_details
+        self.notification = notification
+
+    def build(self) -> dict[str, object]:
+        project = Project.objects.get_from_cache(id=self.group.project_id)
+        event_for_tags = self.event or self.group.get_latest_event()
+        timestamp = (
+            max(self.group.last_seen, self.event.datetime) if self.event else self.group.last_seen
+        )
+        obj: Group | GroupEvent = self.event if self.event is not None else self.group
+        rule_id = None
+        if self.rules:
+            rule_id = self.rules[0].id
+
+        embeds = [
+            DiscordMessageEmbed(
+                title=build_attachment_title(obj),
+                description=build_attachment_text(self.group, self.event) or None,
+                url=get_title_link(
+                    self.group,
+                    self.event,
+                    self.link_to_event,
+                    self.issue_details,
+                    self.notification,
+                    ExternalProviders.DISCORD,
+                    rule_id,
+                ),
+                color=LEVEL_TO_COLOR["info"],
+                # We can't embed urls in Discord embed footers.
+                footer=DiscordMessageEmbedFooter(
+                    build_footer(self.group, project, self.rules, "{text}")
+                ),
+                fields=build_tag_fields(event_for_tags, self.tags),
+                timestamp=timestamp,
+            )
+        ]
+
+        components = build_components(self.group, project)
+
+        return self._build(embeds=embeds, components=components)
+
+
+def build_tag_fields(
+    event_for_tags: GroupEvent | None, tags: set[str] | None = None
+) -> list[DiscordMessageEmbedField]:
+    fields: list[DiscordMessageEmbedField] = []
+    if tags:
+        event_tags = event_for_tags.tags if event_for_tags else []
+        for key, value in event_tags:
+            std_key = tagstore.get_standardized_key(key)  # type: ignore
+            if std_key not in tags:
+                continue
+
+            labeled_value = tagstore.get_tag_value_label(key, value)  # type: ignore
+            fields.append(
+                DiscordMessageEmbedField(
+                    std_key,
+                    labeled_value,
+                    inline=True,
+                )
+            )
+    return fields
+
+
+def build_components(
+    group: Group,
+    project: Project,
+) -> list[DiscordMessageComponent]:
+
```

Comment:
Please double check this logic makes sense! I mostly copied Slack, with the exception of removing the escalating feature flag from the logic.